### ISSUE TITLE
Server assumes endpoints are always in "endpoints" directory

### ISSUE THREAD

**Issue by penberg** [chiselstrike/chiselstrike#1650]
There's logic in the server that differentiates between endpoints and other types of modules by checking if code is in `endpoints` directory or not. As @honzasp points out, this is wrong: users can configure endpoints to reside in a different directory.

**Comment by honzasp** [comment]
This particular code in `chiseld` is a hack and we should rewrite it, but in my opinion we should not allow the user to change the names of these directories (let alone use multiple directories!). Nobody uses this ability, nobody tests for it (so it does not work), it increases implementation complexity and I there is no clear motivation for this feature.

**Comment by penberg** [comment]
@honzasp I am in favour of ripping out the configurability. However, the server should _not_ be looking at the paths to determine the type of a module, I think. Instead, the RPC protocol should have that information, sent by the client.

**Comment by honzasp** [comment]
Indeed! If we have merged #1618, the problem in `chiseld` would be solved...
